File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_text(string text, int array[]);
int colemanLiau(float l, float s);

int main(void)
{


    string text = get_string("Text: ");

    int arrayNumbersOfText[3];
    int arr = count_text(text, arrayNumbersOfText);


    float leterrPerWords = 100 * ((float) arrayNumbersOfText[2] / (float) arrayNumbersOfText[1]);
    float wrodsPerSentences = 100 * ((float) arrayNumbersOfText[0] / (float)arrayNumbersOfText[1]);

    /* Printfs */
    //printf("%d letters\n", arrayNumbersOfText[2]);
    //printf("%d words\n", arrayNumbersOfText[1]);
    //printf("%d sentences\n", arrayNumbersOfText[0]);

    int colemanValue = colemanLiau(leterrPerWords, wrodsPerSentences);

    if (colemanValue >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (colemanValue < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", colemanValue);
    }

}

int count_text(string text, int array[])
{
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            //ADD sentences
            array[0] += 1;
        }
        if (text[i] == 32)
        {
            //ADD words
            array[1] += 1;
        }
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            //ADD letters
            array[2] += 1;
        }


    }
    array[1] += 1;
    return 0;
}

int colemanLiau(float l, float s)
{
    float index;



    index = (0.0588 * l) - (0.296 * s) - 15.8;

    index = round(index);
    return index;
}

File: sentimental-readability/readability.py
# TODO
import cs50
def main():
    text = getText()
    colemanValue = colemanLiau(text)
    printResult(colemanValue)


def getText():
    text = cs50.get_string("Text: ")
    return text


def colemanLiau(text):

    setencesWordsLetters = countSetencesWordsLetters(text)

    numberSentences = setencesWordsLetters[0]
    numberWords = setencesWordsLetters[1]
    numberLetters = setencesWordsLetters[2]

    L =(float(numberLetters)/float(numberWords)*100)
    S =(float(numberSentences)/float(numberWords)*100)


    return ((0.0588 * L) - (0.296 * S) - 15.8)



def countSetencesWordsLetters(arrayWords):

    setencesWordsLetters = [0,1,0]

    for word in arrayWords:
        for letter in word:
            if (ord(letter) == 33 or ord(letter) == 46 or ord(letter) == 63):
                setencesWordsLetters[0] +=1

            if (ord(letter) == 32):
                setencesWordsLetters[1] +=1

            if ((ord(letter) > 64 and ord(letter) < 91) or (ord(letter) > 96 and ord(letter) < 123)):
                setencesWordsLetters[2] +=1

    return setencesWordsLetters

def printResult(colemanValue):

    if colemanValue >= 16:
        print("Grade 16+")
    elif colemanValue < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(colemanValue)}")


main()